driver-core/src/test/java/com/datastax/driver/core/querybuilder/QueryBuilderTest.java

```diff
@@ -865,24 +864,65 @@ public void should_quote_column_names_with_escaped_quotes() {
-        assertThat(select.getValues()).isNull();
+        assertThat(select.getValueDefinitions()).isEmpty();
     }
 
-    @Test(groups = "unit", expectedExceptions = { IllegalStateException.class })
-    public void should_throw_ISE_if_getObject_called_on_statement_without_values() {
+    @Test(groups = "unit", expectedExceptions = { IllegalArgumentException.class })
+    public void should_throw_IAE_if_getObject_called_on_statement_without_values() {
         builder.select().from("test").where(eq("foo", 42)).getObject(0); // integers are appended to the CQL string
     }
 
-    @Test(groups = "unit", expectedExceptions = { IndexOutOfBoundsException.class })
-    public void should_throw_IOOBE_if_getObject_called_with_wrong_index() {
-        builder.select().from("test").where(eq("foo", new Object())).getObject(1);
+    @Test(groups = "unit", expectedExceptions = { IllegalArgumentException.class })
+    public void should_throw_IAE_if_getObject_called_with_wrong_index() {
+        builder.select().from("test").where(eq("foo", new Date())).getObject(1);
     }
 
     @Test(groups = "unit")
     public void should_return_object_at_ith_index() {
-        Object expected = new Object();
+        Object expected = new Date();
         Object actual = builder.select().from("test").where(eq("foo", expected)).getObject(0);
-        assertThat(actual).isSameAs(expected);
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test(groups = "unit")
+    @CassandraVersion(major = 2.1)
+    public void should_serialize_collections_of_serializable_elements() {
+        Set<Integer> set = Sets.newHashSet(1, 2, 3);
+        List<Integer> list = Lists.newArrayList(1, 2, 3);
+        Map<Integer, String> map = ImmutableMap.of(1, "foo");
+        BuiltStatement query = builder.insertInto("foo").value("v", set);
+        assertThat(query.getQueryString()).isEqualTo("INSERT INTO foo (v) VALUES (?);");
+        assertThat(query.getObject(0)).isEqualTo(set);
+        query = builder.insertInto("foo").value("v", list);
+        assertThat(query.getQueryString()).isEqualTo("INSERT INTO foo (v) VALUES (?);");
+        assertThat(query.getObject(0)).isEqualTo(list);
+        query = builder.insertInto("foo").value("v", map);
+        assertThat(query.getQueryString()).isEqualTo("INSERT INTO foo (v) VALUES (?);");
+        assertThat(query.getObject(0)).isEqualTo(map);
+    }
+
+    @Test(groups = "unit")
+    @CassandraVersion(major = 2.1)
+    public void should_not_attempt_to_serialize_function_calls_in_collections() {
+        BuiltStatement query = builder.insertInto("foo").value("v", Sets.newHashSet(fcall("func", 1)));
+        assertThat(query.getQueryString()).isEqualTo("INSERT INTO foo (v) VALUES ({func(1)});");
+        assertThat(query.getValueDefinitions()).isNullOrEmpty();
+    }
+
+    @Test(groups = "unit")
+    @CassandraVersion(major = 2.1)
+    public void should_not_attempt_to_serialize_bind_markers_in_collections() {
```

Comment:
I realized recently when merging JAVA-983 that the query generated by this test is invalid, because bind markers are not allowed in collection literals: you need to put one single bind marker for the whole collection. This test has thus been deleted.